db/data/20230907045337_disable_users_in_migrated_facilities.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class DisableUsersInMigratedFacilities < ActiveRecord::Migration[6.1]
+  REGIONS_WITH_ONGOING_ACCESS = [
+    {region_type: :state, slug: "west-bengal"},
+    {region_type: :state, slug: "nagaland"},
+    {region_type: :district, slug: "chennai"},
+    {region_type: :district, slug: "pimpri-chinchwad-municipal-corporation"},
+    {region_type: :district, slug: "pune-municipal-corporation"}
+  ]
+  def up
+    unless CountryConfig.current_country?("India") && ENV["SIMPLE_SERVER_ENV"] == "production"
+      return print "DisableUsersInMigratedFacilities is only for production India"
+    end
+
+    users_with_remaining_access = REGIONS_WITH_ONGOING_ACCESS.flat_map do |region_details|
+      region = Region.find_by(region_details)
+      next unless region
+
+      if region.district_region?
+        region.source.users.sync_approval_status_allowed
+      elsif region.state_region?
+        region.children.flat_map { |district| district.source.users.sync_approval_status_allowed }
+      end
+    end
+
+    all_allowed_users = User.all.sync_approval_status_allowed.pluck(:id)
+    disabled_users = all_allowed_users - users_with_remaining_access.map(&:id)
```

Comment:
[nit] could pluck again? not that important here though.
```suggestion
    disabled_users = all_allowed_users - users_with_remaining_access.pluck(:id)
```

Reply:
Can't use pluck here since `users_with_remaining_access` is an array and not an active record relation.

Reply:
ActiveSupport defines `pluck` on all Enumerables these days :)
https://api.rubyonrails.org/classes/Enumerable.html#method-i-pluck